File: backend/core-service/src/main/java/com/ner/logistics/vehicle/Vehicle.java
package com.ner.logistics.vehicle;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "vehicles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String code; // e.g. NER-07

    private String licensePlate;

    private String vehicleType; // e.g. Emergency Truck, Fuel Tanker

    private Double capacityTons;

    private Long driverId;

    @Column(nullable = false)
    private String status; // ON_TRACK, DELAYED, AT_RISK
}

File: apps/mobile-app/lib/shared/models/vehicle.dart
class Vehicle {
  final int? id;
  final String code;
  final String? licensePlate;
  final String? vehicleType;
  final double? capacityTons;
  final String status;

  Vehicle({
    this.id,
    required this.code,
    this.licensePlate,
    this.vehicleType,
    this.capacityTons,
    required this.status,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'],
      code: json['code'] ?? '',
      licensePlate: json['licensePlate'],
      vehicleType: json['vehicleType'],
      capacityTons: (json['capacityTons'] as num?)?.toDouble(),
      status: json['status'] ?? 'ON_TRACK',
    );
  }
}
